packages/plugin-typescript/README.md

```diff
@@ -0,0 +1,167 @@
+# @code-pushup/typescript-plugin
+
+[![npm](https://img.shields.io/npm/v/%40code-pushup%2Ftypescript-plugin.svg)](https://www.npmjs.com/package/@code-pushup/typescript-plugin)
+[![downloads](https://img.shields.io/npm/dm/%40code-pushup%2Ftypescript-plugin)](https://npmtrends.com/@code-pushup/typescript-plugin)
+[![dependencies](https://img.shields.io/librariesio/release/npm/%40code-pushup/typescript-plugin)](https://www.npmjs.com/package/@code-pushup/typescript-plugin?activeTab=dependencies)
+
+🕵️ **Code PushUp plugin for measuring TypeScript quality with compiler diagnostics.** 🔥
+
+This plugin allows you to **incrementally adopting strict compilation flags in TypeScript projects**.
+It analyzes your codebase using the TypeScript compiler to detect potential issues and configuration problems.
+
+TypeScript compiler diagnostics are mapped to Code PushUp audits in the following way:
+
+- `value`: The number of issues found for a specific TypeScript configuration option -> 3
+- `displayValue`: The number of issues found -> 3 issues
+- `score`: Binary scoring - 1 if no issues are found, 0 if any issues exist
+- Issues are mapped to audit details, containing:
+  - Source file location
+  - Error message from TypeScript compiler
+  - Code reference where the issue was found
+
+## Getting started
+
+1. If you haven't already, install [@code-pushup/cli](../cli/README.md) and create a configuration file.
+
+2. Install as a dev dependency with your package manager:
+
+   ```sh
+   npm install --save-dev @code-pushup/typescript-plugin
+   ```
+
+   ```sh
+   yarn add --dev @code-pushup/typescript-plugin
+   ```
+
+   ```sh
+   pnpm add --save-dev @code-pushup/typescript-plugin
+   ```
+
+3. Add this plugin to the `plugins` array in your Code PushUp CLI config file (e.g. `code-pushup.config.ts`).
+
+Define the ts config file used to compile your codebase. Based on those compiler options the plugin will generate audits.
+
+```ts
+import typescriptPlugin from '@code-pushup/typescript-plugin';
+
+export default {
+  // ...
+  plugins: [
+    // ...
+    typescriptPlugin({
+      tsConfigPath: './tsconfig.json',
+      onlyAudits: ['no-implicit-any'],
+    }),
+  ],
+};
+```
```

Comment:
Please indent all blocks under the list item (3 spaces).

```md
3. Add this plugin to the `plugins` array in your Code PushUp CLI config file (e.g. `code-pushup.config.ts`).

   Define the ts config file used to compile your codebase. Based on those compiler options the plugin will generate audits.

   ```ts
   import typescriptPlugin from '@code-pushup/typescript-plugin';

   // ... etc. ...
```